standalone-dns-proxy/cmd/root.go

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"log/slog"
+	"os"
+
+	"github.com/cilium/hive/cell"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/cilium/cilium/pkg/fqdn/service"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+var (
+	// StandaloneDNSProxyCell provides the standalone DNS proxy functionality
+	// It is responsible for starting and stopping the standalone DNS proxy
+	// based on the configuration provided in the config. It connects to the
+	// cilium agent via gRPC to receive DNS rules and send DNS responses.
+	StandaloneDNSProxyCell = cell.Module(
+		"standalone-dns-proxy",
+		"Provides the standalone DNS proxy functionality",
+
+		cell.Provide(func() *option.DaemonConfig {
+			return option.Config
+		}),
+		cell.Provide(func() service.FQDNConfig {
+			return service.DefaultConfig
+		}),
+		cell.Invoke(registerStandaloneDNSProxyHooks),
+	)
+
+	binaryName = "standalone-dns-proxy"
+)
+
+func NewDNSProxyCmd(h *hive.Hive) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   binaryName,
+		Short: "Run " + binaryName,
+		Run: func(cobraCmd *cobra.Command, args []string) {
+			logger := logging.DefaultSlogLogger.With(logfields.LogSubsys, binaryName)
+
+			initEnv(logger, h.Viper())
+
+			if err := h.Run(logging.DefaultSlogLogger); err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+	h.RegisterFlags(cmd.Flags())
+
+	cmd.AddCommand(
+		h.Command(),
+	)
+	cobra.OnInitialize(option.InitConfig(logging.DefaultSlogLogger, cmd, "Standalone-DNS-Proxy", "standalone-dns-proxy", h.Viper()))
```

Comment:
How far off are we from not having to pull in option.Config? It'd be a nice bit of cleanup if we can move FQDN related options into their own config struct and out from option.DaemonConfig.

Reply:
So we would still need to pull `option.Config`, but I tried to move some config out.
As i create more PR for sdp, I'll try to move more config to fqdn.